app.js

```diff
@@ -35,14 +35,28 @@ var budgetController = (function() {
     }
 
     function getDeleteButtonDivInfo (elementId) {
-
+        var myDiv = document.getElementById(elementId);
+        var itemValue = myDiv.querySelector('.item__value').textContent.slice(2);
+        var plusOrMinus = myDiv.querySelector('.item__value').textContent.substring(0, 1);
+
+        if (plusOrMinus == '+') { 
+            totalIncome = totalIncome - parseFloat(itemValue);
+        } else { 
+            totalExpenses = totalExpenses - parseFloat(itemValue);
+        }
+        availableBudget = totalIncome - totalExpenses;
+        expensePercentOfTotalIncome = Math.floor(totalExpenses / totalIncome) * 100;
+        if (plusOrMinus == '+') {
+            allExpenses = document.querySelectorAll('[id^="expense-"]');

```

Comment:
I need to redeclare `allExpenses` on line 50 in order to use .forEach on line 51.